### PR TITLE
[#1991] re-organizing infinite scroll for conversationlist and messagelist

### LINKED ISSUE #1991: Re-organizing Infinite Scroll for ConversationList and MessageList
**Current state**
- we have an architecture problem regarding infinite scrolling for messages and conversations 
- the way we pass props is also problematic: we pass too much data to the components, we need to fine-grain the way we pass props

**To do** 
- investigate ways we can improve the infinite loading of messages and conversations as well as the way we pass props

### PR DESCRIPTION
**Pull request by AudreyKj**
closes #1991 (follow up to #1679)

---> implementation of the conversation list from scratch (fixes the flickering bug #1679]
---> added react.memo to messageList to prevent flickering on click 
---> fine-grained props passing in messageList

### CODE REVIEW

File: frontend/ui/src/pages/Inbox/Messenger/MessageList/index.tsx [hunk 1]
```diff
@@ -25,9 +26,14 @@ type MessageListProps = ConnectedProps<typeof connector> & {
...
+  console.log('ownProps', ownProps.match.params.conversationId);
   return {
     messages: getCurrentMessages(state, ownProps),
-    conversation: getConversation(state, ownProps),
+    conversationId: ownProps.match.params.conversationId,
+    conversationChannelSource: state.data.conversations.all.items[ownProps.match.params.conversationId].channel.source,
+    conversationMetadataContact:
+      state.data.conversations.all.items[ownProps.match.params.conversationId].metadata.contact,
+    conversationPaginationData: state.data.conversations.all.items[ownProps.match.params.conversationId].paginationData,
   };
```
Comment (AudreyKj) [inline frontend/ui/src/pages/Inbox/Messenger/MessageList/index.tsx:37]: here reduced the props passed to MessageList to the necessary ones

File: frontend/ui/BUILD [hunk 2]
```diff
@@ -32,8 +32,6 @@ ts_web_library(
         "@npm//lodash-es",
         "@npm//react",
         "@npm//react-router-dom",
-        "@npm//react-window",
-        "@npm//react-window-infinite-loader",
```
Comment (AudreyKj) [inline frontend/ui/BUILD:36]: removing all unnecessary dependencies here

File: frontend/ui/src/pages/Inbox/ConversationList/index.tsx [hunk 3]
```diff
@@ -114,7 +91,24 @@ const ConversationList = (props: ConversationListProps) => {
...
+                filteredConversations.map(conversation => (
+                  <ConversationListItem
+                    key={conversation.id}
+                    conversation={conversation}
+                    active={conversation.id === currentConversationId}
+                  />
+                ))}
+            </>
+          )}
+        </div>
```
Comment (AudreyKj) [inline frontend/ui/src/pages/Inbox/ConversationList/index.tsx:110]: implementation from scratch of conversation list

**Review by AitorAlgorta — APPROVED** [review]

**Pull request merged**